format/src/main/thrift/carbondata.thrift

```diff
@@ -183,11 +183,23 @@ struct FileFooter{
 }
 
 /**
+* Footer for indexed carbon file
+*/
+struct FileFooterVersion3{
+    1: required i64 num_rows; // Total number of rows in this file
+    2: required SegmentInfo segment_info;	// Segment info (will be same/repeated for all files in this segment)
+    3: required list<BlockletIndex> blocklet_index_list;	// blocklet index of all blocklets in this file
+    4: optional list<BlockletInfo3> blocklet_info_list3;	// Information about blocklets of all columns in this file
+    5: optional dictionary.ColumnDictionaryChunk dictionary; // blocklet local dictionary
+}
+
+/**
  * Header for appendable carbon file
  */
 struct FileHeader{
 	1: required i32 version; // version used for data compatibility
 	2: required list<schema.ColumnSchema> table_columns;  // Description of columns in this file
+	3: required bool is_Footer_Present; //  to check whether footer is present or not      
```

Comment:
Making the field as `required` will loose backward compatability. Better to make as `optional`